testutil/beaconmock/options.go

```diff
@@ -149,6 +149,12 @@ func WithValidatorSet(set ValidatorSet) Option {
 				}
 			}
 
+			if len(resp) == 0 {
```

Comment:
move this to the top, so first check if "querying for all", in that case return all, else pick from set and log errs.